react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { current: 0 };
+    this.imgs = [];
+    this.progressDots = [];
+    this.handleClick = this.handleClick.bind(this);
+    this.interval = setInterval(() => {
+      this.setState({ current: (this.state.current + 1) % this.imgs.length });
+    }, 3000);
+  }
+
+  handleClick(event) {
+    if (event.target.matches('i')) {
+      clearInterval(this.interval);
+      this.interval = setInterval(() => {
+        this.setState({ current: (this.state.current + 1) % this.imgs.length });
+      }, 3000);
+      if (event.target.matches('.fa-chevron-left')) {
+        this.setState({ current: (this.state.current - 1) < 0 ? this.imgs.length - 1 : this.state.current - 1 });
+      } else if (event.target.matches('.fa-chevron-right')) {
+        this.setState({ current: (this.state.current + 1) % this.imgs.length });
+      } else if (event.target.matches('.progress-circle')) {
+        this.setState({ current: Number.parseInt(event.target.getAttribute('index')) });
+      }
+    }
+  }
```

Comment:
This is fine, but sometimes it is easier to make multiple methods and attach them to their corresponding elements. For example you could have the following methods: `nextImg()`, `previousImg()`, and `setImg(imgId)`. This approach would simplify your logic and make your code more maintainable.

Reply:
I'll do that for future assignments, thanks for the advice !